pyproject.toml

```diff
@@ -3,7 +3,7 @@ requires = [
     "Cython>=0.29.23",
     # Building against NPY2 will support both NPY1 and NPY2
     # https://numpy.org/devdocs/dev/depending_on_numpy.html#build-time-dependency
-    "numpy>=2.0.1",
```

Comment:
Not sure we pin `>=2.0.1` for any reason, but [numpy manual](https://numpy.org/devdocs/dev/depending_on_numpy.html#build-time-dependency) seems to recommended `2.0.0rc1`:
> https://numpy.org/devdocs/dev/depending_on_numpy.html#build-time-dependency